File: researchland.js
const fetch = require('node-fetch');

// Define the base URL for land listings API (replace with your actual API URL)
const LAND_API_URL = 'https://example.com/land-listings-api';

async function fetchLandListings(location, maxPrice, minSize) {
    try {
        // Define query parameters
        const queryParams = new URLSearchParams({
            location,
            maxPrice,
            minSize,
        });

        // Build the complete API URL
        const apiUrl = `${LAND_API_URL}?${queryParams}`;

        // Send a GET request to the land listings API
        const response = await fetch(apiUrl);

        // Check the response status
        if (!response.ok) {
            throw new Error(`HTTP error! Status: ${response.status}`);
        }

        // Parse JSON response
        const landListings = await response.json();

        return landListings;
    } catch (error) {
        console.error(`An error occurred: ${error.message}`);
        return [];
    }
}

function displayLandListings(landListings) {
    if (!landListings.length) {
        console.log('No matching land listings found.');
        return;
    }

    console.log('Matching Land Listings:');
    landListings.forEach(listing => {
        console.log(`Location: ${listing.location}`);
        console.log(`Price: $${listing.price}`);
        console.log(`Size: ${listing.size} acres`);
    });
}

async function main() {
    console.log('Land Research and Location Program');
    const location = prompt('Enter location: ');
    const maxPrice = parseFloat(prompt('Enter maximum price ($): '));
    const minSize = parseFloat(prompt('Enter minimum land size (acres): '));

    const landListings = await fetchLandListings(location, maxPrice, minSize);
    displayLandListings(landListings);
}

main();

File: researchland.rb
require 'net/http'
require 'json'

# Define the base URL for land listings API (replace with your actual API URL)
LAND_API_URL = 'https://example.com/land-listings-api'

def fetch_land_listings(location, max_price, min_size)
  begin
    # Define query parameters
    query_params = {
      location: location,
      max_price: max_price,
      min_size: min_size
    }

    # Build the complete API URL
    api_url = URI("#{LAND_API_URL}?#{URI.encode_www_form(query_params)}")

    # Send a GET request to the land listings API
    response = Net::HTTP.get(api_url)

    # Parse JSON response
    land_listings = JSON.parse(response)

    return land_listings
  rescue StandardError => e
    puts "An error occurred: #{e.message}"
    return []
  end
end

def display_land_listings(land_listings)
  if land_listings.empty?
    puts 'No matching land listings found.'
  else
    puts 'Matching Land Listings:'
    land_listings.each do |listing|
      puts "Location: #{listing['location']}"
      puts "Price: $#{listing['price']}"
      puts "Size: #{listing['size']} acres"
    end
  end
end

def main
  puts 'Land Research and Location Program'
  print 'Enter location: '
  location = gets.chomp
  print 'Enter maximum price ($): '
  max_price = gets.to_f
  print 'Enter minimum land size (acres): '
  min_size = gets.to_f

  land_listings = fetch_land_listings(location, max_price, min_size)
  display_land_listings(land_listings)
end

main
